strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/io/MavenMetadataExpiredRepositoryPathHandler.java

```diff
@@ -35,16 +35,10 @@ public boolean supports(final RepositoryPath repositoryPath)
         {
             return false;
         }
-        try
-        {
-            if (!RepositoryFiles.isMetadata(repositoryPath))
-            {
-                return false;
-            }
-        }
-        catch (IOException e)
+        boolean isMetadata = ThrowingFunction.unchecked(RepositoryFiles::isMetadata).apply(repositoryPath);
```

Comment:
I think we can add `throws IOException` for `ExpiredRepositoryPathHandler.supports` method and use `ThrowingPredicate` where this method used

Reply:
@adavid9 you don't need to catch and re-throw `IOException` here, just add it to method declaration